lib/_tls_wrap.js

```diff
@@ -444,9 +444,7 @@ TLSSocket.prototype.renegotiate = function(options, callback) {
   }
   if (!this._handle.renegotiate()) {
     if (callback) {
-      process.nextTick(function() {
-        callback(new Error('Failed to renegotiate'));
-      });
+      process.nextTick(callback, new Error('Failed to renegotiate'));
```

Comment:
Of note, this will change the stacktrace for the Error. Not sure if that's a *huge* problem, since the new stack will be slightly more descriptive.